test/e2e/agent_manager.go

```diff
@@ -0,0 +1,283 @@
+	}
+
+	// Build the agent
+	buildCmd := exec.CommandContext(m.ctx, "go", "build", "-o", "agent", "./cmd/agent")
+	buildCmd.Dir = m.agentPath
+	buildCmd.Stdout = os.Stdout
+	buildCmd.Stderr = os.Stderr
+
+	if err := buildCmd.Run(); err != nil {
+		return fmt.Errorf("failed to build agent: %w", err)
+	}
+
+	fmt.Println("rhobs-synthetics-agent built successfully.")
+	return nil
+}
+
+// createConfig creates the agent configuration file
+func (m *AgentManager) createConfig() error {
+	configContent := fmt.Sprintf(`---
+api_urls:
+  - %s/probes
+label_selector: "app=rhobs-synthetics-probe"
+polling_interval: 2s
+log_level: debug
+log_format: text
+`, m.apiURL)
+
+	m.configPath = filepath.Join(os.TempDir(), "agent-config.yaml")
+	if err := os.WriteFile(m.configPath, []byte(configContent), 0644); err != nil {
+		return fmt.Errorf("failed to write config file: %w", err)
+	}
+
+	return nil
+}
+
+// startAgent starts the agent process
+func (m *AgentManager) startAgent() error {
+	binaryPath := filepath.Join(m.agentPath, "agent")
+	
+	// Agent uses Cobra with "start" subcommand
+	m.cmd = exec.CommandContext(m.ctx, binaryPath, "start", "--config", m.configPath)
+	m.cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	// Capture stdout and stderr
+	stdout, err := m.cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get stdout pipe: %w", err)
+	}
+
+	stderr, err := m.cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get stderr pipe: %w", err)
+	}
+
+	// Start the process
+	if err := m.cmd.Start(); err != nil {
+		return fmt.Errorf("failed to start agent process: %w", err)
+	}
+
+	// Stream logs in background (filter out noise)
+	go m.streamLogs(stdout, "STDOUT")
+	go m.streamLogs(stderr, "STDERR")
+
+	// Wait a bit for agent to start
+	time.Sleep(2 * time.Second)
+
+	// Check if process is still running
+	if m.cmd.Process == nil {
+		return fmt.Errorf("agent process failed to start")
+	}
+
+	return nil
+}
+
+// streamLogs streams logs from the agent
+func (m *AgentManager) streamLogs(reader io.Reader, prefix string) {
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		
+		// Filter to only show important agent activity
+		// Show: startup, probe processing, status updates, errors
+		// Skip: routine reconciliation cycles, fetching counts, prometheus messages
+		if containsAny(line, []string{
+			"Agent started with config",
+			"Processing probe",
+			"Successfully updated probe",
+			"Successfully processed probe",
+			"Failed to create Kubernetes",
+			"error", "ERROR", "Error:",
+			"Probe .* processed",
+		}) {
+			// Skip overly verbose source info in JSON logs
+			if strings.Contains(line, `"source":`) {
+				// Parse JSON and extract just time, level, and msg
+				var logEntry map[string]interface{}
+				if err := json.Unmarshal([]byte(line), &logEntry); err == nil {
+					if msg, ok := logEntry["msg"].(string); ok {
+						level := "INFO"
```

Comment:
Should this be configurable for troubleshooting, or at least set as a constant at the top?